### PR TITLE
set up accessors for Series.dt and also fix to_series()

### LINKED ISSUE #131: Properties on `.dt` accessor no longer work as of `1.4.3.220704`
**Describe the bug**

After upgrading to `1.4.3.220704`, I receive the following errors:

```python
from typing import TYPE_CHECKING

import pandas as pd

if not TYPE_CHECKING:
    try:
        from typing import reveal_type
    except ImportError:
        from typing_extensions import reveal_type

i0 = pd.date_range(start="2022-06-01", periods=10, tz="UTC")

reveal_type(i0)  # note: Revealed type is "pandas.core.indexes.datetimes.DatetimeIndex"
# Runtime type is 'DatetimeIndex'
reveal_type(i0.to_series())  # note: Revealed type is "Any"
# Runtime type is 'Series'

s0 = pd.Series(i0)

reveal_type(s0)  # note: Revealed type is "pandas.core.series.TimestampSeries"
# Runtime type is 'Series'
reveal_type(s0.dt)  # note: Revealed type is "pandas.core.indexes.accessors.CombinedDatetimelikeProperties"
# Runtime type is 'DatetimeProperties'

s0.dt.date  # error: "CombinedDatetimelikeProperties" has no attribute "date"  [attr-defined]
s0.dt.time  # error: "CombinedDatetimelikeProperties" has no attribute "time"  [attr-defined]
s0.dt.timetz  # error: "CombinedDatetimelikeProperties" has no attribute "timetz"  [attr-defined]
s0.dt.year  # error: "CombinedDatetimelikeProperties" has no attribute "year"  [attr-defined]
s0.dt.month  # error: "CombinedDatetimelikeProperties" has no attribute "month"  [attr-defined]
s0.dt.day  # error: "CombinedDatetimelikeProperties" has no attribute "day"  [attr-defined]
s0.dt.hour  # error: "CombinedDatetimelikeProperties" has no attribute "hour"  [attr-defined]
s0.dt.minute  # error: "CombinedDatetimelikeProperties" has no attribute "minute"  [attr-defined]
s0.dt.second  # error: "CombinedDatetimelikeProperties" has no attribute "second"  [attr-defined]
s0.dt.microsecond  # error: "CombinedDatetimelikeProperties" has no attribute "microsecond"  [attr-defined]
s0.dt.nanosecond  # error: "CombinedDatetimelikeProperties" has no attribute "nanosecond"  [attr-defined]
s0.dt.week  # error: "CombinedDatetimelikeProperties" has no attribute "week"  [attr-defined]
s0.dt.weekofyear  # error: "CombinedDatetimelikeProperties" has no attribute "weekofyear"  [attr-defined]
s0.dt.dayofweek  # error: "CombinedDatetimelikeProperties" has no attribute "dayofweek"  [attr-defined]
s0.dt.day_of_week  # error: "CombinedDatetimelikeProperties" has no attribute "day_of_week"  [attr-defined]
s0.dt.weekday  # error: "CombinedDatetimelikeProperties" has no attribute "weekday"  [attr-defined]
s0.dt.dayofyear  # error: "CombinedDatetimelikeProperties" has no attribute "dayofyear"  [attr-defined]
s0.dt.day_of_year  # error: "CombinedDatetimelikeProperties" has no attribute "day_of_year"  [attr-defined]
s0.dt.quarter  # error: "CombinedDatetimelikeProperties" has no attribute "quarter"  [attr-defined]
s0.dt.is_month_start  # error: "CombinedDatetimelikeProperties" has no attribute "is_month_start"  [attr-defined]
s0.dt.is_month_end  # error: "CombinedDatetimelikeProperties" has no attribute "is_month_end"  [attr-defined]
s0.dt.is_quarter_start  # error: "CombinedDatetimelikeProperties" has no attribute "is_quarter_start"  [attr-defined]
s0.dt.is_quarter_end  # error: "CombinedDatetimelikeProperties" has no attribute "is_quarter_end"  [attr-defined]
s0.dt.is_year_start  # error: "CombinedDatetimelikeProperties" has no attribute "is_year_start"  [attr-defined]
s0.dt.is_year_end  # error: "CombinedDatetimelikeProperties" has no attribute "is_year_end"  [attr-defined]
s0.dt.is_leap_year  # error: "CombinedDatetimelikeProperties" has no attribute "is_leap_year"  [attr-defined]
s0.dt.daysinmonth  # error: "CombinedDatetimelikeProperties" has no attribute "daysinmonth"  [attr-defined]
s0.dt.days_in_month  # error: "CombinedDatetimelikeProperties" has no attribute "days_in_month"  [attr-defined]
s0.dt.tz  # error: "CombinedDatetimelikeProperties" has no attribute "tz"  [attr-defined]
s0.dt.freq  # OK
s0.dt.isocalendar()  # error: "CombinedDatetimelikeProperties" has no attribute "isocalendar"  [attr-defined]
s0.dt.to_period("D")  # error: "CombinedDatetimelikeProperties" has no attribute "to_period"  [attr-defined]
s0.dt.to_pydatetime()  # OK
# s0.dt.tz_localize("UTC")  # error: "CombinedDatetimelikeProperties" has no attribute "tz_localize"  [attr-defined]
s0.dt.tz_convert("UTC")  # error: "CombinedDatetimelikeProperties" has no attribute "tz_convert"  [attr-defined]
s0.dt.normalize()  # error: "CombinedDatetimelikeProperties" has no attribute "normalize"  [attr-defined]
s0.dt.strftime("%Y")  # error: "CombinedDatetimelikeProperties" has no attribute "strftime"  [attr-defined]
s0.dt.round("D")  # error: "CombinedDatetimelikeProperties" has no attribute "round"  [attr-defined]
s0.dt.floor("D")  # error: "CombinedDatetimelikeProperties" has no attribute "floor"  [attr-defined]
s0.dt.ceil("D")  # error: "CombinedDatetimelikeProperties" has no attribute "ceil"  [attr-defined]
s0.dt.month_name()  # error: "CombinedDatetimelikeProperties" has no attribute "month_name"  [attr-defined]
s0.dt.day_name()  # error: "CombinedDatetimelikeProperties" has no attribute "day_name"  [attr-defined]

i1 = pd.period_range(start="2022-06-01", periods=10)

reveal_type(i1)  # note: Revealed type is "pandas.core.indexes.datetimes.PeriodIndex"
# Runtime type is 'PeriodIndex'
reveal_type(i1.to_series())  # note: Revealed type is "Any"
# Runtime type is 'Series'

s1 = pd.Series(i1)

reveal_type(s1)  # note: Revealed type is "pandas.core.series.Series[Any]"
# Runtime type is 'Series'
reveal_type(s1.dt)  # note: Revealed type is "pandas.core.indexes.accessors.CombinedDatetimelikeProperties"
# Runtime type is 'PeriodProperties'

s1.dt.qyear  # error: "CombinedDatetimelikeProperties" has no attribute "qyear"  [attr-defined]
s1.dt.start_time  # error: "CombinedDatetimelikeProperties" has no attribute "start_time"  [attr-defined]
s1.dt.end_time  # error: "CombinedDatetimelikeProperties" has no attribute "end_time"  [attr-defined]

i2 = pd.timedelta_range(start="1 day", periods=10)

reveal_type(i2)  # note: Revealed type is "pandas.core.indexes.datetimes.TimedeltaIndex"
# Runtime type is 'TimedeltaIndex'
reveal_type(i2.to_series())  # note: Revealed type is "Any"
# Runtime type is 'Series'

s2 = pd.Series(i2)

reveal_type(s2)  # note: Revealed type is "pandas.core.series.Series[Any]"
# Runtime type is 'Series'
reveal_type(s2.dt)  # note: Revealed type is "pandas.core.indexes.accessors.CombinedDatetimelikeProperties"
# Runtime type is 'TimedeltaProperties'

s2.dt.days  # error: "CombinedDatetimelikeProperties" has no attribute "days"  [attr-defined]
s2.dt.seconds  # error: "CombinedDatetimelikeProperties" has no attribute "seconds"  [attr-defined]
s2.dt.microseconds  # error: "CombinedDatetimelikeProperties" has no attribute "microseconds"  [attr-defined]
s2.dt.nanoseconds  # error: "CombinedDatetimelikeProperties" has no attribute "nanoseconds"  [attr-defined]
s2.dt.components  # OK
s2.dt.to_pytimedelta()  # OK
s2.dt.total_seconds()  # error: "CombinedDatetimelikeProperties" has no attribute "total_seconds"  [attr-defined]
```

This is likely due to [this change](https://github.com/pandas-dev/pandas-stubs/commit/4641fa8e84b2bca394db0753acd4663866abf73c#diff-6f90f15b437974706a3c0a75636efd6d163a42d4a786bc5513a5069b1040f031R810). I guess the problem is that `.dt` previously resolved to `Series[Any]` and thus all of the properties/methods resolved to `Any`.

**To Reproduce**
1. Provide a minimal runnable `pandas` example that is not properly checked by the stubs.
2. Indicate which type checker you are using (`mypy` ~or  `pyright`~).
3. Show the error message received from that type checker while checking your example.

**Please complete the following information:**
 - OS: Linux
 - OS Version: N/A
 - python version: `3.9` and `3.10`
 - version of type checker: `0.961`
 - version of installed `pandas-stubs`: `1.4.3.220704`

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by Dr-Irv**
- [x] Closes #131(Replace xxxx with the Github issue number)
- [x] Tests added: Please use `assert_type()` to assert the type of any return value
    - `test_timefuncs.py` - `test_dt_accessors()`

It would take a bit of work to do `check(assert_type(series.dt.xxx, ...))` for all of those `Series.dt.xxx` calls, so we could leave that for a future issue, but at least we would pass the tests that came from the original issue for now.

### CODE REVIEW

File: pandas-stubs/core/indexes/accessors.pyi [hunk 1]
```diff
@@ -1,28 +1,239 @@
...
+)
 from pandas.core.base import (
-    NoNewAttributesMixin as NoNewAttributesMixin,
-    PandasObject as PandasObject,
+    NoNewAttributesMixin,
+    PandasObject,
+)
+from pandas.core.frame import DataFrame
+from pandas.core.series import (
+    Series,
```
Comment (twoertwein) [inline pandas-stubs/core/indexes/accessors.pyi:22]: Could shorten it to `from pandas import DataFrame, Series`

Comment (twoertwein) [reply pandas-stubs/core/indexes/accessors.pyi:22]: Probably better keep as-is, would be more important for the tests.

**Review by twoertwein — APPROVED** [review]
Looks good to me! Would be nice to assert the return type with `assert_type` (and `check`) but could be added later.

**Comment by Dr-Irv** [comment]
> Looks good to me! Would be nice to assert the return type with `assert_type` (and `check`) but could be added later.

I realized overnight that the types of all the accessors are wrong - they need to all be `Series`, so I will add the checks.

**Comment by Dr-Irv** [comment]
@twoertwein have added all the checks and `assert_type` in the tests, so now I'm more confident that the results are correct.  Asking again for your review.

File: pandas-stubs/core/series.pyi [hunk 2]
```diff
@@ -155,6 +161,16 @@ class Series(IndexOpsMixin, NDFrame, Generic[S1]):
         fastpath: bool = ...,
     ) -> TimestampSeries: ...
     @overload
+    def __new__(
+        cls,
+        data: PeriodIndex,
+        index: Optional[Union[_str, int, Series, List, Index]] = ...,
+        dtype=...,
+        name: Optional[Hashable] = ...,
```
Comment (twoertwein) [inline pandas-stubs/core/series.pyi:169]: Could be shortened to `Hashable`. `None` is `Hashable`. Could also keep it to be more explicit.

Comment (Dr-Irv) [reply pandas-stubs/core/series.pyi:169]: > Could be shortened to `Hashable`. `None` is `Hashable`. Could also keep it to be more explicit.

For consistency, I think I will keep it to be more explicit.  Not sure that people know that `None` is hashable.

**Review by twoertwein — APPROVED** [review]
Thank you, for asserting all of the types! I honestly, find it more difficult to review these changes compared to the inline annotations in pandas as I don't have the code/doc-string around - but adding all the test helps!!

**Comment by twoertwein** [comment]
Thank you @Dr-Irv!

**Pull request merged**